### PR TITLE
fix: bring backup wallet out of sub menu

### LINKED ISSUE #3829: Bring backup wallet out from sub menu for native wallet
### Overview

We would like to bring the backup wallet button out from the sub menu as it is currently buried.

### References and additional details

This is all in the <UserMenu /> component

### Acceptance Criteria

Backup my wallet will be on the first menu and not buried in a sub menu

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by reallybeard**
## Description

- Wallets can now add a `connectedMenuComponent` that will render within the wallet dropdown
- Native right now is the only wallet that will use this
- I also tidied up the KeepKey menu a little

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #3829 

## Risk

small risk

### Operations

When connected to a native wallet you will now see in the wallet dropdown a "backup my wallet" menu item. And the header should be in a disabled state to signify that it has no sub menu.

This should not change the currently functionality for existing wallet sub menus e.g. KeepKey

## Screenshots (if applicable)
Screen Shot 2023-02-14 at 10 55 02 AM
Screen Shot 2023-02-14 at 10 56 39 AM

### CODE REVIEW

File: src/context/WalletProvider/config.ts [hunk 1]
```diff
@@ -75,6 +75,7 @@ export interface SupportedWalletInfo {
   routes: RouteProps[]
   connectedWalletMenuRoutes?: RouteProps[]
   connectedWalletMenuInitialPath?: WalletConnectedRoutes
+  connectedMenuComponent?: React.ComponentType<any> | undefined
```
Comment (0xApotheosis) [inline src/context/WalletProvider/config.ts:78]:
```suggestion
  connectedMenuComponent?: React.ComponentType<any>
```

File: src/components/Layout/Header/NavBar/UserMenu.tsx [hunk 2]
```diff
@@ -148,18 +150,13 @@ export const UserMenu: React.FC<{ onClick?: () => void }> = ({ onClick }) => {
...
-        <MenuButton
-          as={IconButton}
-          aria-label='Open wallet dropdown menu'
-          icon={<ChevronDownIcon />}
+        <WalletButton
+          onConnect={handleConnect}
+          walletInfo={walletInfo}
+          isConnected={isConnected}
+          isDemoWallet={isDemoWallet}
+          isLoadingLocalWallet={state.isLoadingLocalWallet}
```
Comment (0xApotheosis) [inline src/components/Layout/Header/NavBar/UserMenu.tsx:159]: Might as well spread this guy on line 143, and then

```suggestion
          isLoadingLocalWallet={isLoadingLocalWallet}
```

**Review by 0xApotheosis — COMMENTED** [review]
Found one issue where the `connectedMenuComponent` is always highlighted (selected) upon opening the menu, which feels a bit wierd:

Screenshot 2023-02-15 at 6 14 25 am

Otherwise a couple of trivial comments.

**Comment by 0xdef1cafe** [comment]
smol KK visual regressions on hover state

before (prod) 

image

after (local)

image

**Comment by 0xdef1cafe** [comment]
native looks 🛍️ though

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**